README.adoc

```diff
@@ -0,0 +1,68 @@
+# microprofile.config.incubator
+//
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+# Configuration for Java EE
+
+## Status
+
+Config proposal!
+
+== Rational
+
+Typically, majority applications need to be configured based on running environment. The configuration data need to be stored in a different location so that the applications should not need to be repackaged. The configuration data (data source) can come from different locations and in different formats (e.g. system properties, system environment variables, .properties, .yml, datasource). If a property is defined in multiple data sources, which value should be used. Therefore, there should be a policy to order the data sources. Under some circumstances, some data sources may change dynamically. The changed value should be fed into the client without the need of restarting the application. This requirement is particular important for micro services running in a cloud environment.
+
+== Implementaions
```

Comment:
Is the proposed API based on one of the implementations? Is any of them already compliant with the proposed API? I think this should be clear from the beginning, so that we may discuss further which way to go and if any of them could be used as a reference impl.